### PR TITLE
fix: Display positioning and sizing was wrong

### PR DESCRIPTION
**Pull request by wayfarer3130**
### Context

The display area settings weren't fully functional when used near the edges, or with rotation.  This fixes the various bugs with the display area settings.

### Changes & Results

Make rotation work around the getPan center point
Make display area zoom 100% images actually zoom to 100%
Make panning calculations for display area actual pan to edge of viewport

### Testing

Run the examples in yarn example stackPosition

#### PR

- [] My Pull Request title is descriptive, accurate and follows the
  semantic-release format and guidelines.

#### Code

- [] My code has been well-documented (function documentation, inline comments,
  etc.)

- [] I have run the `yarn build:update-api` to update the API documentation, and have
  committed the changes to this PR. (Read more here https://www.cornerstonejs.org/docs/contribute/update-api)

#### Public Documentation Updates

- [] The documentation page has been updated as necessary for any public API
  additions or removals.

#### Tested Environment

- [] "OS: 
- [] "Node version: 
- [] "Browser:
  

[blog]: https://circleci.com/blog/triggering-trusted-ci-jobs-on-untrusted-forks/
[script]: https://github.com/jklukas/git-push-fork-to-upstream-branch

### CODE REVIEW

File: packages/core/examples/stackPosition/index.ts [hunk 1]
```diff
@@ -0,0 +1,269 @@
...
+// This is for debugging purposes
+console.warn(
+  'Click on index.ts to open source code for this example --------->'
+);
+
+const { ViewportType, Events } = Enums;
+
+// ======== Constants ======= //
+const renderingEngineId = 'myRenderingEngine';
+const viewportId = 'CT_STACK';
```
Comment (sedghi) [inline packages/core/examples/stackPosition/index.ts:27]: trying the example, clicking on the top left change the VOI https://deploy-preview-930--cornerstone-3d-docs.netlify.app/live-examples/stackposition

File: packages/core/src/RenderingEngine/Viewport.ts [hunk 2]
```diff
@@ -584,14 +584,15 @@ class Viewport implements IViewport {
...
 
     const { imageArea, imageCanvasPoint } = displayArea;
 
+    let zoom = 1;
     if (imageArea) {
       const [areaX, areaY] = imageArea;
-      const zoom = Math.min(this.getZoom() / areaX, this.getZoom() / areaY);
-      this.setZoom(zoom, storeAsInitialCamera);
+      zoom = Math.min(this.getZoom() / areaX, this.getZoom() / areaY);
+      this.setZoom(1.1 * zoom, storeAsInitialCamera);
```
Comment (sedghi) [inline packages/core/src/RenderingEngine/Viewport.ts:595]: what is 1.1? please add a comment

File: packages/core/examples/stackPosition/index.ts [hunk 3]
```diff
@@ -0,0 +1,269 @@
...
+    const viewport = <Types.IStackViewport>(
+      renderingEngine.getViewport(viewportId)
+    );
+
+    // Set a range to highlight bones
+    viewport.setProperties({ voiRange: { upper: 2500, lower: -1500 } });
+
+    viewport.render();
+  },
+});
```
Comment (sedghi) [inline packages/core/examples/stackPosition/index.ts:169]: rename or even remove

File: packages/core/examples/stackPosition/index.ts [hunk 4]
```diff
@@ -0,0 +1,269 @@
...
+    },
+  };
+}
+
+const displayAreas = new Map();
+displayAreas.set('Center with border', createDisplayArea(1.1, 0.5));
+displayAreas.set('Center Full', createDisplayArea(1, 0.5));
+displayAreas.set('Left Top', createDisplayArea(1, 0));
+displayAreas.set('Right Top', createDisplayArea(1, [1, 0]));
+displayAreas.set('Left Bottom', createDisplayArea(1, [0, 1]));
```
[the remark this answers is not in the document]
Comment (wayfarer3130) [reply packages/core/examples/stackPosition/index.ts:113]: There is already a resized example further down.  Yes, they are supposed to do the same thing in this situation, but that actually needs testing.

[the remark this answers is not in the document]
Comment (wayfarer3130) [reply packages/core/examples/stackPosition/index.ts:113]: They should do the same thing regardless of viewport size - that is actually part of what is being demonstrated.

**Review by sedghi — APPROVED** [review]

**Pull request merged**